### ISSUE TITLE
qwen2-vl error on inference - system/assistant prompts not being processed properly

### ISSUE THREAD

**Issue by cedonley** [matatonic/openedai-vision#19]
I get an error if I include a system prompt (or assistant messages).

Was able to solve by changing qwen2-vl.py. Not sure if this is the best fix, but it seems to work after this. Note that there are sometimes multiple contents in a system prompt (especially in the qwen-agent tools examples). Seems they can just be concatenated, but unsure if they should remain as a list like the user messages...

```
@@ -51,7 +53,10 @@ class VisionQnA(VisionQnABase):
                         msg['content'].extend([{'type': c.type, 'video': c.imag
e_url.url}])
             else:
                 #msg = { 'role': m.role, 'content': [{ 'type': 'text', 'text':
c.text }] }
-                msg = { 'role': m.role, 'content': c.text }
+                mtext = ""
+                for c in m.content:
+                    mtext += c.text
+                msg = { 'role': m.role, 'content': [{'type': 'text', 'text': mt
ext}]}
```

**Comment by matatonic** [comment]
Oh, interesting. I've never seen multiple contents in a system prompt. I'll test some options and fix, thanks!

**Comment by cedonley** [comment]
This may be related to using qwen-agent, which seems to include an additional content if you include tools.

But, I'm also seeing that if I don't change the content to include the list of dicts (as above) vs just a string, that I get an error when using flash attention 2. The above seems to fix this issue as well.

**Comment by matatonic** [comment]
did you try "\n".join() instead of append?

**Comment by matatonic** [comment]
ie. 
```python
                ctext = "\n".join([c.text for c in m.content])
                msg = { 'role': m.role, 'content': [{ 'type': 'text', 'text': ctext }] }
```

**Comment by cedonley** [comment]
When the qwen-agent examples send multiple contents, the second content (w/ tools strings) starts with "\n\n". My guess is you can probably just run a loop like you do on the user role. In fact, maybe you don't even need to differentiate between different roles. I didn't get a chance to try it yet.

**Comment by matatonic** [comment]
If it's not much bother, is there a way you can dump the final context which is sent to the API, maybe with DEBUG enabled on the server.

**Comment by matatonic** [comment]
oh, do you mean that the real fix may just be using this format:

msg = { 'role': m.role, 'content': [{ 'type': 'text', 'text': c.text }] }

**Comment by matatonic** [comment]
No, I just checked the tokenizer chat messages and it expects the first and only messages to be a system, nothing else will get processed. So single message, first, is correct.

**Comment by cedonley** [comment]
Ok. So based on that, it wants it concatenated before tokenization.

I had printed debug logs earlier, but away from that computer at the moment. When I debugged, the second content started with "\n\n", so the right way to join it would be straight concatenation of the text into a single string without adding additional "\n".

The examples I used are the two VL examples in the qwen-agent repository's examples folder if you have a moment to try them yourself.
